src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/DateTimeValueResolver.php

```diff
@@ -74,6 +85,9 @@ public function resolve(Request $request, ArgumentMetadata $argument): array
             }
             try {
                 $date = new $class($value);
+                if ($this->clock) {
+                    $date = $this->clock->withTimeZone($date->getTimezone())->now()->modify($value);
```

Comment:
Why using the clock here when providing an explicit value ? I don't think `$date->modify('2022-12-18')` works fine.

and this does not respect `$class` anymore.

Reply:
I try to support relative date time (ex: `-3 hours`). We can decide that's not supported when a clock is used.

The syntax `$date->modify('2022-12-18')` works, but not exactly how I expected. https://3v4l.org/YRRDr#v8.1.12

Reverting this change.